worlds/smw/__init__.py

```diff
@@ -50,78 +51,89 @@ class SMWWorld(World):
     lost all of his abilities. Can he get them back in time to save the Princess?
     """
     game: str = "Super Mario World"
-    option_definitions = smw_options
+
     settings: typing.ClassVar[SMWSettings]
+
+    options_dataclass = SMWOptions
+    options: SMWOptions
+
     topology_present = False
-    data_version = 3
-    required_client_version = (0, 3, 5)
+    required_client_version = (0, 4, 4)
```

Comment:
This is largely a question for my own understanding, but worth pointing out here anyway, I suppose:

The next AP version won't ship as version 0.4.4, and we all know this when developing clients that get shipped with AP. But an actual 0.4.4 install can connect to a server running a game that used this version of the world to generate, and AP won't reject it. So if we merge this and immediately ship the next AP version, incompatible clients will be allowed to connect to new slots. But in trying to predict the next release number during one of these PRs you make it impossible to connect your client on the current commit to a game generated on the current commit (and can just be wrong about the version number).

Do we have a solution for this? Is the expectation just to play catch-up with your world's `required_client_version` between releases by making a PR to bump it between the AP version bump and the first RC?